src/p5xr/core/raycasting.js

```diff
@@ -0,0 +1,127 @@
+function getRayFromScreen(screenX, screenY) {
+  let ray = {
+    origin: new p5.Vector(0, 0, 0),
+    direction: new p5.Vector()
+  };
+
+  let poseMatrix = p5xr.instance.viewer.poseMatrix.copy();
+  poseMatrix.transpose(poseMatrix);
+  poseMatrix = poseMatrix.mat4;
+
+  // set origin of ray to pose position
+  ray.origin.x = poseMatrix[3];
+  ray.origin.y = poseMatrix[7];
+  ray.origin.z = poseMatrix[11];
+  
+  let initialMVMatrix = p5xr.instance.viewer.initialMVMatrix.copy();
+  initialMVMatrix.transpose(initialMVMatrix);
+  initialMVMatrix = initialMVMatrix.mat4;
+
+  
+  // transform ray origin to view space
+  let rayOriginCopy = ray.origin.copy();
+  ray.origin.x = initialMVMatrix[0] * rayOriginCopy.x + initialMVMatrix[1] * rayOriginCopy.y + initialMVMatrix[2] * rayOriginCopy.z + initialMVMatrix[3];
+  ray.origin.y = initialMVMatrix[4] * rayOriginCopy.x + initialMVMatrix[5] * rayOriginCopy.y + initialMVMatrix[6] * rayOriginCopy.z + initialMVMatrix[7];
+  ray.origin.z = initialMVMatrix[8] * rayOriginCopy.x + initialMVMatrix[9] * rayOriginCopy.y + initialMVMatrix[10] * rayOriginCopy.z + initialMVMatrix[11];
+
+  // get ray direction from left eye
+  let leftDirection = new p5.Vector(screenX, screenY, -1);
+
+  let leftPMatrixInverse = new p5.Matrix();
+  leftPMatrixInverse.invert(p5xr.instance.viewer.leftPMatrix.copy());
+  leftPMatrixInverse.transpose(leftPMatrixInverse);
+  leftPMatrixInverse = leftPMatrixInverse.mat4;
+
+  let leftDirectionCopy = leftDirection.copy();
+  leftDirection.x = leftPMatrixInverse[0] * leftDirectionCopy.x + leftPMatrixInverse[1] * leftDirectionCopy.y + leftPMatrixInverse[2] * leftDirectionCopy.z + leftPMatrixInverse[3];
+  leftDirection.y = leftPMatrixInverse[4] * leftDirectionCopy.x + leftPMatrixInverse[5] * leftDirectionCopy.y + leftPMatrixInverse[6] * leftDirectionCopy.z + leftPMatrixInverse[7];
+  leftDirection.normalize();
+
+  // get ray direction from right eye
+  let rightDirection = new p5.Vector(screenX, screenY, -1);
+  
+  let rightPMatrixInverse = new p5.Matrix();
+  rightPMatrixInverse.invert(p5xr.instance.viewer.rightPMatrix.copy());
+  rightPMatrixInverse.transpose(rightPMatrixInverse);
+  rightPMatrixInverse = rightPMatrixInverse.mat4;
+
+  let rightDirectionCopy = rightDirection.copy();
+  rightDirection.x = rightPMatrixInverse[0] * rightDirectionCopy.x + rightPMatrixInverse[1] * rightDirectionCopy.y + rightPMatrixInverse[2] * rightDirectionCopy.z + rightPMatrixInverse[3];
+  rightDirection.y = rightPMatrixInverse[4] * rightDirectionCopy.x + rightPMatrixInverse[5] * rightDirectionCopy.y + rightPMatrixInverse[6] * rightDirectionCopy.z + rightPMatrixInverse[7];
+  rightDirection.normalize();
+
+  // combine both ray directions
+  ray.direction = p5.Vector.add(leftDirection, rightDirection).normalize();
+
+  return ray;
+}
+
+p5.prototype.intersectsSphere = function(radius, arg2, arg3) {
```

Comment:
I am not sure if I agree with the `arg2`, `arg3` naming convention. I understand why you would try to be less prescriptive about what they are but I don't know if there is a good reason here.

Reply:
I named like that because arg2 can mean screenX or ray object. Anyways, I am now using `arguments` to capture the parameters.